packages/garbo/src/tasks/barfTurn.ts

```diff
@@ -340,6 +342,56 @@ const NonBarfTurnTasks: AlternateTask[] = [
     spendsTurn: true,
     choices: { 1091: 7 },
   },
+  {
+    name: "Apriling Saxophone Lucky (drunk)",
+    completed: () => !AprilingBandHelmet.canPlay("Apriling band saxophone"),
+    ready: () =>
+      globalOptions.ascend &&
+      have($item`Apriling band saxophone`) &&
+      getBestLuckyAdventure().phase === "barf" &&
+      getBestLuckyAdventure().value() > get("valueOfAdventure"),
+    do: () => getBestLuckyAdventure().location,
+    prepare: () => {
+      if (!have($effect`Lucky!`)) {
+        AprilingBandHelmet.play($item`Apriling band saxophone`);
+      }
+    },
+    outfit: () => ({ offhand: $item`Drunkula's wineglass` }),
+    combat: new GarboStrategy(() =>
+      Macro.abortWithMsg("Unexpected combat while attempting Lucky! adventure"),
+    ),
+    turns: () =>
+      have($item`Apriling band saxophone`)
+        ? $item`Apriling band saxophone`.dailyusesleft
+        : 0,
+    sobriety: "drunk",
+    spendsTurn: true,
+  },
+  {
+    name: "Apriling Saxophone Lucky (sober)",
+    completed: () => !AprilingBandHelmet.canPlay("Apriling band saxophone"),
+    ready: () =>
+      !globalOptions.ascend &&
+      have($item`Apriling band saxophone`) &&
+      getBestLuckyAdventure().phase === "barf" &&
+      getBestLuckyAdventure().value() > get("valueOfAdventure"),
+    do: () => getBestLuckyAdventure().location,
+    prepare: () => {
+      if (!have($effect`Lucky!`)) {
+        AprilingBandHelmet.play($item`Apriling band saxophone`);
+      }
+    },
+    outfit: () => ({}),
+    combat: new GarboStrategy(() =>
+      Macro.abortWithMsg("Unexpected combat while attempting Lucky! adventure"),
+    ),
+    turns: () =>
+      have($item`Apriling band saxophone`)
+        ? $item`Apriling band saxophone`.dailyusesleft
+        : 0,
+    sobriety: "sober",
+    spendsTurn: true,
+  },
```

Comment:
instead of repeating all this code I'd either compress them into one task or make a function like we see for daily dungeon